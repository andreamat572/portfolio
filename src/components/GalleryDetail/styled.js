import styled from 'styled-components';

export const StyledGalleryDetail = styled.li`
  border: 1px solid #e2e2e2;
  border-radius: 0.5rem;
  background: #ffffff;

  > figure {
    overflow: hidden;
    border-radius: 0.5rem;
    position: relative;
  }

  > figure > img {
    width: 100%;
  }

  > figure > figcaption {
    position: absolute;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.7);
    width: 100%;
    border-bottom-left-radius: 0.5rem;
    border-bottom-right-radius: 0.5rem;
  }

  > figure > figcaption > h3 {
    color: white;
    padding: 0.75rem;
    font-size: 1.25rem;
  }

  > p {
    font-size: 1rem;
    line-height: 1.5;
    padding: 1rem 0.75rem;
    color: #666666;
  }

  > a {
    padding: 0.5rem 1rem;
    margin: 0.5rem;
  }
`;
