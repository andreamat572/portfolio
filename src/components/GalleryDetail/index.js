import React, { PureComponent } from 'react';
import { StyledGalleryDetail } from './styled';
import Image from '../Image';

export default class GalleryDetail extends PureComponent {
  render() {
    const { image, name } = this.props;
    return (
      <StyledGalleryDetail>
        <figure>
          <Image src={image} alt={image} />
          <figcaption>
            <a href={`/gallery/${name}/${image}`}>
              <h3>{name + ' ' + image}</h3>
            </a>
          </figcaption>
        </figure>
      </StyledGalleryDetail>
    );
  }
}
