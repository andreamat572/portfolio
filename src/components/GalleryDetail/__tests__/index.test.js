import { shallow } from 'enzyme';
import React from 'react';
import GalleryDetail from '../index';

describe('<GalleryDetail/>', () => {
  it('renders correctly', () => {
    const props = {
      image: {
        name: 'fashion1',
        description: 'Bella',
        location: 'Berlin',
        published_at: '2019',
      },
      name: 'gallery1',
      description: 'a description',
      location: 'unknown',
      published_at: '2019',
    };

    const wrapper = shallow(<GalleryDetail {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
