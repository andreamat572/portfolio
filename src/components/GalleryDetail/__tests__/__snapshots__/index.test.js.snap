// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GalleryDetail/> renders correctly 1`] = `ShallowWrapper {}`;
